homeassistant/components/discovergy/sensor.py

```diff
@@ -152,35 +154,86 @@ async def async_setup_entry(
     data: DiscovergyData = hass.data[DOMAIN][entry.entry_id]
     meters: list[Meter] = data.meters  # always returns a list
 
-    entities: list[DiscovergySensor] = []
+    entities: list[DiscovergyBaseSensor] = []
     for meter in meters:
         sensors = None
         if meter.measurement_type == "ELECTRICITY":
             sensors = ELECTRICITY_SENSORS
         elif meter.measurement_type == "GAS":
             sensors = GAS_SENSORS
 
+        coordinator: DiscovergyUpdateCoordinator = data.coordinators[meter.meter_id]
+
         if sensors is not None:
             for description in sensors:
                 # check if this meter has this data, then add this sensor
                 for key in {description.key, *description.alternative_keys}:
-                    coordinator: DiscovergyUpdateCoordinator = data.coordinators[
-                        meter.meter_id
-                    ]
                     if key in coordinator.data.values:
                         entities.append(
                             DiscovergySensor(key, description, meter, coordinator)
                         )
 
+        if coordinator.data.time:
+            entities.append(DiscovergyLastTransmittedTimeSensor(meter, coordinator))
+
     async_add_entities(entities, False)
 
 
-class DiscovergySensor(CoordinatorEntity[DiscovergyUpdateCoordinator], SensorEntity):
+class DiscovergyBaseSensor(
+    CoordinatorEntity[DiscovergyUpdateCoordinator], SensorEntity
+):
+    """Represents a discovergy smart meter base sensor."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        meter: Meter,
+        coordinator: DiscovergyUpdateCoordinator,
+    ) -> None:
+        """Initialize the base sensor."""
+        super().__init__(coordinator)
+
+        self.meter = meter
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, meter.meter_id)},
+            name=f"{meter.measurement_type.capitalize()} {meter.location.street} {meter.location.street_number}",
+            model=f"{meter.type} {meter.full_serial_number}",
+            manufacturer=MANUFACTURER,
+        )
+
+
+class DiscovergyLastTransmittedTimeSensor(DiscovergyBaseSensor):
```

Comment:
Can't we just expand the sensor entity descriptions instead of creating more entity classes?

Reply:
Sure, we can, just thought it would be cleaner to have it separate. Change is coming in a few minutes.